CHANGELOG.md

```diff
@@ -23,6 +23,12 @@
 
 #### Breaking changes
 
+- R-devel Non-API changes:
+  - R's C API is being formalized. While the changes are formalized, non-API functions are hidden behind a feature flag to prevent removal from CRAN.
+  - See [[Rd] clarifying and adjusting the C API for R](https://stat.ethz.ch/pipermail/r-devel/2024-June/083449.html)
+  - [nonAPI.txt](https://github.com/r-devel/r-svn/blob/f36c203d3a53a74d56a81d4f97a68d24993e0652/src/library/tools/R/sotools.R#L564) functions are hidden behind the `non-api` feature flag.
+  - Removed from default include (but may not be limited to):
+    - `global_var()`, `local_var()`, `base_env()`, various `Environment`, `Function`, `Primitive`, and `Promise` methods.
```

Comment:
@CGMossa please review this changelog.

Reply:
Yes, this work. Please mention CRAN somewhere there, because it seems a little absurd that we are pre-emptively pushing R-devel changes to extendr. Of course, it is because CRAN is rejecting packages that don't comply immediately.

Reply:
Yes i've updated it a bit.